pkgs/development/libraries/db/generic.nix

```
@@ -67,8 +67,7 @@ stdenv.mkDerivation (rec {
       (if compat185 then "--enable-compat185" else "--disable-compat185")
     ]
     ++ lib.optional dbmSupport "--enable-dbm"
-    ++ lib.optional stdenv.isFreeBSD "--with-pic"
-    ++ (drvArgs.configureFlags or []);
+    ++ lib.optional stdenv.isFreeBSD "--with-pic";
```

```suggestion
    ++ lib.optionals stdenv.isFreeBSD [ "--with-pic" ];
```
Opinionated: I would use `lib.optionals` instead, since `configureFlags` is a list. Semantically, it makes more sense to me.

---

so this change is just a revert of the previous workaround for pkgsLLVM so i'd prefer to leave as is (it is the first commit). my change is the 2nd commit which modifies the patch. in general, i do try to prefer optionals over optional.

---

It's totally fine and optional, no worries.